source/libNsdl/src/sn_grs.c

```diff
@@ -23,7 +23,7 @@
  */
 #include <string.h>
 #include <stdlib.h>
-
+#include <assert.h>
```

Comment:
True about NDEBUG, but if we were only thinking about NDEBUG, then there's never any issue.

The essence of the problem is "where does the output go", and so if we use tr_xxx rather than printf, we should use tr_error or a new tr_assert rather than assert. (Presumably a release build should also have the trace disabled?)

A secondary question to consider is "what happens after the output"? Standard assert() calls abort(), but what does that do in an embedded system? I believe MBED_ASSERT infinite loops, and a few of our own error handlers do the same.

That feels a bit more like a platform HAL thing, which isn't just covered by the existing trace. We do have a fairly-widely-implemented board reset call, but is halting better?